text/0000-layer-table.md

```diff
@@ -0,0 +1,89 @@
+# Meta
+[meta]: #meta
+- Name: Move layer types to new table in layer.toml
+- Start Date: 2021-01-05
+- Author(s): natalieparellano
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Introduces a new table in `<layers>/<layer>.toml` to enable Bash buildpacks to easily opt in to layer re-use.
+
+# Definitions
+[definitions]: #definitions
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 0052](https://github.com/buildpacks/rfcs/blob/main/text/0052-opt-in-layer-caching.md) described a change (to be implemented in Buildpack API 0.6) whereby buildpacks must opt in to layer re-use - i.e., each `<layers>/<layer>.toml` will be restored with no `launch`, `build`, or `cache` flags set and buildpacks must modify the restored TOML to set these flags.
+
+The RFC assumed that Bash buildpacks would be able to easily opt in to layer re-use by appending lines to the TOML, e.g.:
+
+`echo "launch = true" >> layer.toml`
+
+However, because `launch`, `build`, and `cache` are top level keys, the above command would add a `launch` key to the `[metadata]` table instead of the top level.
```

Comment:
I don't fully get this – can you explain this a bit more deeply? If buildpacks are restricted to writing non-top level keys, then how would they be able to write a top-level types array?

Reply:
After the buildpack makes its edits the file would look like:
```
[metadata]
...stuff...

[types]
launch = true
build = false
cache = false
```

Without `[types]`, `launch = true` etc. would be keys under `[metadata]`.

Reply:
Ahh. Hence prepending would work, because it's assured to be a top level key.